src/app/Mappings/components/MappingBuilder/MappingBuilder.tsx

```diff
@@ -33,7 +33,7 @@ export const MappingBuilder: React.FunctionComponent<IMappingBuilderProps> = ({
   setBuilderItems,
 }: IMappingBuilderProps) => {
   const messageSelectBoth = 'You must select a source and target before adding another mapping.';
-  const messageExhausted = `All sources ${mappingType.toLowerCase()} have been mapped.`;
+  const messageExhausted = `All sources ${mappingType.toLowerCase()}s have been mapped.`;
```

Comment:
@gildub this is still going to produce the wrong spelling/phrasing.
1. `mappingType` is either Network or Storage, we need "networks" and *"datastores"*
2. The word "sources" should not be plural, only the type.

We need this:
```ts
  const messageExhausted = `All source ${
    mappingType === MappingType.Network ? 'networks' : 'datastores'
  } have been mapped.`;
```

Reply:
Sorry, I shouldn't have been working on it late last night, not paying enough attention.

Reply:
No problem at all!

Yeah, it's JS expressions all the way down :)